lib/kafka-producer.js

```diff
@@ -32,6 +33,8 @@ Producer.prototype.getProducer = Promise.method(function (opts, topts) {
     opts['metadata.broker.list'] = this.kafkaBrokerUrl;
   }
 
+  this.__shouldFailWhenSchemaIsMissing = !!shouldFailWhenSchemaIsMissing;
```

Comment:
It's better `shouldFailWhenSchemaIsMissing === true` to avoid any kind of coertion problems

Reply:
Ok, I changed it in https://github.com/waldophotos/kafka-avro/pull/63/commits/95d34d698105e3d58b495c3c1b0086caa76db675